onnxscript/tools/benchmark/export_model_batch.py

```diff
@@ -0,0 +1,140 @@
+        new_args=args,
+    )
+
+    print("-------------------")
+    print("[export_model]")
+    pprint.pprint(kwargs)
+    print("-------------------")
+
+    import openpyxl
+    import pandas
+
+    assert openpyxl
+    from onnxscript.tools.benchmark.benchmark_helpers import (
+        BenchmarkError,
+        run_benchmark,
+    )
+
+    script_name = "onnxscript.tools.benchmark.export_model"
+
+    configs: list[dict[str, Any]] = [
+        dict(exporter="eager"),
+        dict(ort_optimize=1, exporter="script"),
+        dict(ort_optimize=1, optimization="optimize,rewrite,inline", exporter="script"),
+        dict(ort_optimize=0, optimization="optimize,rewrite,inline", exporter="script"),
+        dict(ort_optimize=1, optimization="", exporter="dynamo"),
+        dict(ort_optimize=1, optimization="optimize,rewrite,inline", exporter="dynamo"),
+        dict(ort_optimize=0, optimization="optimize,rewrite,inline", exporter="dynamo"),
+    ]
+    common_kwargs: dict[str, Any] = kwargs.copy()
+    common_kwargs["verbose"] = max(common_kwargs["verbose"] - 1, 0)
+    for c in configs:
+        c.update(common_kwargs)
+
+    if kwargs["verbose"]:
+        for i, cf in enumerate(configs):
+            print(f"[export_common_batch] config {i+1}: {cf}")
+
+    ################################
+    # Running configuration.
+
+    try:
+        data = run_benchmark(
+            script_name,
+            configs,
+            verbose=kwargs["verbose"],
+            stop_if_exception=False,
+        )
+        data_collected = True
+    except BenchmarkError as e:
+        if kwargs["verbose"]:
+            print(e)
+        data_collected = False
+
+    prefix = "_".join(
+        [
+            "emb_",
+            kwargs["model"],
+            "dynamic" if kwargs["dynamic"] else "static",
+            kwargs["dtype"].replace("float", "fp"),
+            kwargs["device"],
+            kwargs["config"],
+            f"h{kwargs['num_hidden_layers']}",
+        ],
+    )
+
+    if data_collected:
+        df = pandas.DataFrame(data)
+        df = df.drop(["OUTPUT", "ERROR"], axis=1)
+        df["repeat_time"] = df["repeat_time"].astype(float)
+        df_eager = df[(df["implementation"] == "eager") & (df["exporter"] == "eager")][
+            "repeat_time"
+        ].dropna()
+        if df_eager.shape[0] > 0:
+            min_eager = df_eager.min()
+            df["increase"] = df["repeat_time"] / min_eager - 1
+        filename = f"{prefix}_with_cmd.csv"
+        df.to_csv(filename, index=False)
+
+        df = df.drop(["CMD"], axis=1)
+        filename = f"{prefix}.csv"
+        df.to_csv(filename, index=False)
+        df = pandas.read_csv(filename)  # to cast type
+        print(df)
+
+        # summary
+        cs = [
+            c
+            for c in ["exporter", "optimization", "warmup_time", "repeat_time", "increase"]
+            if c in df.columns
+        ]
+        dfs = df[cs]
+        filename = f"{prefix}_summary.xlsx"
```

Comment:
Is xlsx needed or would csv be sufficient?

Reply:
When I click on a csv file from excel, it always opens it as a single column document. Maybe because it a French setting and , is the decimal separator in France. So I usually export both. Maybe somebody has a better trick...